Apps/W1/HybridGP/app/src/Migration/Support/HelperFunctions.codeunit.al

```diff
@@ -1473,46 +1567,55 @@ codeunit 4037 "Helper Functions"
     var
         GPPaymentTerms: Record "GP Payment Terms";
         PaymentTerms: Record "Payment Terms";
+        GPMigrationLog: Record "GP Migration Log";
         DueDateCalculation: DateFormula;
         DiscountDateCalculation: DateFormula;
         SeedValue: integer;
         PaymentTerm: Text[10];
         DueDateCalculationText: Text[50];
         DiscountDateCalculationText: Text[50];
+        LogMigrationArea: Text[50];
+        IsPaymentTermHandled: Boolean;
     begin
+        LogMigrationArea := 'Payment Terms';
         SeedValue := 0;
         if GPPaymentTerms.FindSet() then begin
             repeat
-                if StrLen(DelChr(GPPaymentTerms.PYMTRMID, '>', ' ')) > 10 then begin
-                    PaymentTerm := GeneratePaymentTerm(SeedValue, GPPaymentTerms.PYMTRMID);
-                    PaymentTerms.Validate(Code, PaymentTerm);
-                    SeedValue := SeedValue + 1;
+                IsPaymentTermHandled := false;
+                OnHandlePaymentTerm(GPPaymentTerms, IsPaymentTermHandled);
+                if not IsPaymentTermHandled then begin
```

Comment:
Minor it is good to extract to method, it would help with the readabiltiy. We can exit the method if it is handled.

Reply:
I feel more comfortable leaving this as-is for now.